README.md

```diff
@@ -2,64 +2,94 @@
+    },
+}
 ```
 
-Creating HTTP request from CloudEvent:
-```
-marshaller := v02.NewDefaultHTTPMarshaller()
-var req *http.Request
-err := marshaller.ToRequest(req)
+Sending a cloudevents.Event via the Http Transport with Binary v0.2 encoding:
+
+```go
+c, err := client.NewHttpClient(context.TODO(), "http://localhost:8080/", cloudeventshttp.BinaryV02)
 if err != nil {
-	panic("Unable to marshal event into http Request: " + err.String())
+	panic("unable to create cloudevent client: " + err.Error())
+}
+if err := c.Send(event); err != nil {
+	panic("failed to send cloudevent: " + err.Error())
 }
 ```
 
-The goal of this package is to provide support for all released versions of CloudEvents, ideally while maintaining
-the same API. It will use semantic versioning with following rules:
-* MAJOR version increments when backwards incompatible changes is introduced.
-* MINOR version increments when backwards compatible feature is introduced INCLUDING support for new CloudEvents version.
-* PATCH version increments when a backwards compatible bug fix is introduced.
-
+Checkout the sample [sender](./cmd/samples/sender) and 
+[receiver](./cmd/samples/receiver) applications for working demo. 
 
 ## TODO list
 
-- [ ] Add encoders registry, where SDK user can register their custom content-type encoders/decoders
-- [ ] Add more tests for edge cases
+### General 
+
+- [ ] increase `./pkg` code coverage to > 90%. (70% as of Feb 19, 2019)
+- [ ] Most tests are happy path, add sad path tests (edge cases).
+- [ ] Use contexts to override internal defaults.
+- [ ] Fill in Event.Context defaults with values (like ID and time) if 
+      nil/empty.
+- [x] Might be nice to have the client have a Receive hook.
+- [ ] Might be an issue with zero body length requests.
+- [ ] Need a change to the client to make making events easier
+- [ ] Implement String() for event context
+
+### Webhook
+- [ ] Implement Auth in webhook
+- [ ] Implement Callback in webhook
+- [ ] Implement Allowed Origin
+- [ ] Implement Allowed Rate
+
+### JSON
+- [ ] Support json value as body. 
+
+### HTTP
+- [ ] Support overrides for method.
+- [ ] Merge headers from context on send.
+
+### Nats
+- [ ] Plumb in auth for the nats server.
+- [ ] v0.2 and v0.3 are very similar. Combine decode logic?
+
+### For v3
+- [ ] Support batch json
 
 ## Existing Go for CloudEvents
 
-Existing projects that added support for CloudEvents in Go are listed below. It's our goal to identify existing patterns
-of using CloudEvents in Go-based project and design the SDK to support these patterns (where it makes sense).
+Existing projects that added support for CloudEvents in Go are listed below. 
+It's our goal to identify existing patterns of using CloudEvents in Go-based 
+project and design the SDK to support these patterns (where it makes sense).
 - https://github.com/knative/pkg/tree/master/cloudevents
 - https://github.com/vmware/dispatch/blob/master/pkg/events/cloudevent.go
```

Comment:
We can probably remove this entry, at least until dispatch can be updated with all of these changes.